include/sparrow/array/typed_array.hpp

```diff
@@ -31,6 +31,28 @@
 
 namespace sparrow
 {
+
+    // forward declaration
+    template <is_arrow_base_type T, arrow_layout L>
+    class typed_array_impl;
+
+    /*
+     * is_typed_array_impl traits
+     */
+    template <class A>
+    struct is_typed_array_impl : std::false_type
+    {
+    };
+
+    template <class T, class L>
+    struct is_typed_array_impl<typed_array_impl<T, L>> : std::true_type
+    {
+    };
+
+    template <class A>
+    constexpr bool is_typed_array_impl_v = is_typed_array_impl<A>::value;
```

Comment:
I suspect this could be implemented using `mpl::is_type_instance_of/_v` ([see there](https://github.com/man-group/sparrow/blob/main/include/sparrow/utils/mp_utils.hpp#L32-L46)) as this is essentially `is_type_instance_of_v<typed_array_impl<T,L>>` ?
I see it's just existing code moved here so not necessarilly tweaked in this pr, as you prefer.

Reply:
I just moved it up because its needed in the requirements for the "from-range" constructor.
We need to make sure the range-constructor is not called for typed-arrays themself